js/script.js

```diff
@@ -27,25 +30,42 @@ const addTask = () => {
 
     const taskTitle = stripSanitizedParts(title);
 
-    const newTask = {
+    if (editingTaskId !== null) {
+        const editedTaskIndex = tasks.findIndex(task => task.id === editingTaskId);
+        if (editedTaskIndex !== -1) {
+            tasks[editedTaskIndex].title = taskTitle;
+            renderTasks();
+            closeForm();
+            return;
+        }
```

Comment:
I believe you have added same functionality inside editTask function. Please do the edit functionality in one place and follow `DRY` convention.  If you need same functionality in multiple place you can make the function reusable and use them as per your requirement.